xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -126,6 +164,277 @@ define(["jquery"], function ($) {
 
 
 
+
+  /**
+   * Return the property descriptor corresponding to a property id
+   * @param {String} propertyId
+   * @returns {Object}
+   */
+  Logic.prototype.getPropertyDescriptor = function (propertyId) {
+    var propertyDescriptor = null;
+    // Unfortunately the Array.prototype.find method is not supported in IE :(
+    this.data.meta.propertyDescriptors.some(function (descriptor) {
+    if (descriptor.id === propertyId) {
+        propertyDescriptor = descriptor;
+        return true;
+      }
+    });
+    return propertyDescriptor;
+  };
+
+
+
+  /**
+   * Return a new displayer based on the specified property and row data
+   * @param {String} propertyId The id of the property of the entry
+   * @param {Object} entry The entry data object
+   * @param {String} displayerId The
+   */
+  Logic.prototype.createDisplayer = function (propertyId, entry, displayerId) {
+    var self = this;
+    var defer = $.Deferred();
+
+    // default displayerId
+    if (displayerId === undefined) {
+      displayerId = (this.getPropertyDescriptor(propertyId).displayer || {}).id || "default";
+    }
+
+    // load displayer based on it's id
+    require([BASE_PATH + "displayers/" + displayerId + "Displayer.js"],
+      // load success
+      function (Displayer) {
+        var displayer = new Displayer(propertyId, entry, self);
+        defer.resolve(displayer);
+      },
+
+      // load failure
+      function (err) {
+        // try to load default layout instead
+        if (displayerId !== "default") {
+          self.createDisplayer(propertyId, entry, "default").then(function (displayer) {
+            defer.resolve(displayer);
+          }, function () {
+            defer.reject();
+          });
+        } else {
+          console.error(err);
+          defer.reject();
+        }
+      });
+
+      return defer.promise();
+  };
+
+
+
+
+  /**
+   * Update sort configuration based on parameters, then fetch new data
+   * @param {String} property The property to sort according to
+   * @param {String} level The sort level of the property (0 is the highest).
+   *   Undefined means current. Negative value removes property sort.
+   * @param {String} descending Specify whether the sort should be descending or not.
+   *   Undefined means toggle current direction
+   */
+  Logic.prototype.sort = function (property, level, descending) {
+    if (this.data.query.properties.indexOf(property) === -1) { return; }
+    // find property current sort level
+    var currentLevel = -1;
+    this.data.query.sort.some(function (sortObject, i) {
```

Comment:
Can't we use polyfills in the meantime?
We could check if `Array.prototype.find` is defined, and if not we replace it by the polyfill. We could define the needed polyfills on top of the file, or in a separate file so that it could be reused, and like that this part of the code wouldn't need to be updated.

Reply:
> Can't we use polyfills in the meantime?

For sure. If it's needed in multiple files then I'd create a polyfill.js file (RequireJS module) and add it as dependency to all the modules that need it.